### ISSUE TITLE
list index out of range while merging multiple time series

### ISSUE THREAD

**Issue by angadsingh** [graphite-project/graphite-web#1493]
We are continuously getting this error (when REMOTE_STORE_MERGE_RESULTS=True, and there are 2 servers both CLUSTERED_SERVER'ed with each other):

`RequestError (http://$server/render/?format=json&from=1459023523&target=aliasByNode%28scale%28prod.%2A.some-query-1.app.%2A.query.dedupe.%2A.%2A.m1_rate%2C+60%29%2C+1%2C+4%2C+7%2C+8%29&until=1459024123): Get failed: 500 INTERNAL SERVER ERROR
Traceback (most recent call last):
  File &quot;/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py&quot;, line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File &quot;/opt/graphite/webapp/graphite/render/views.py&quot;, line 125, in renderView
    seriesList = evaluateTarget(requestContext, target)
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 10, in evaluateTarget
    result = evaluateTokens(requestContext, tokens)
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 21, in evaluateTokens
    return evaluateTokens(requestContext, tokens.expression)
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 28, in evaluateTokens
    args = [evaluateTokens(requestContext, arg) for arg in tokens.call.args]
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 21, in evaluateTokens
    return evaluateTokens(requestContext, tokens.expression)
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 28, in evaluateTokens
    args = [evaluateTokens(requestContext, arg) for arg in tokens.call.args]
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 21, in evaluateTokens
    return evaluateTokens(requestContext, tokens.expression)
  File &quot;/opt/graphite/webapp/graphite/render/evaluator.py&quot;, line 24, in evaluateTokens
    return fetchData(requestContext, tokens.pathExpression)
  File &quot;/opt/graphite/webapp/graphite/render/datalib.py&quot;, line 448, in fetchData
    if j is None and ts[i] is not None:
IndexError: list index out of range"`

The error doesn't come after running the query after a while. Only for fresh data (last 15 minutes).

**Comment by srhopkins** [comment]
Do the metrics exist on both nodes in the cluster? I encountered this on upgrade to 9.15 where we also added 2 more nodes to the hash ring taking advantage of the new merge feature. Most of my errors were actually from storage retention sizes being different where the metrics were living but after fixing those I still get this error on my `scaleToSeconds` metrics. Not sure what it is yet I'm guessing `scale` is an issue somehow.

**Comment by angadsingh** [comment]
Our storage retention size were the same. Just noticed that the storage-aggregation settings were different for these metrics. One was sum with xFilesFactor = 0.1 and the other was average with xFilesFactor = 0.5. But why would it fail the merge (that too for recent data only)?

**Comment by srhopkins** [comment]
I checked today and the ones with issues after the resizing did have different storage-aggregation settings. In my case, inspecting a few metrics with whisper-info I found one would have a 10 secondsPerPoint archive that was missing in the other which had 60 secondsPerPoint. Without officially tracing anything I would guess the one node was returning 6x in its values list/array causing an `IndexError`. Not sure about your situation unless maybe you were sending less frequently around the create timeframe and missing the percent necessary to perform the aggregation but it seems even in that case the arrays would be the same length and just contain more `Nones`.
